sfs/mono/source.py

```diff
@@ -189,31 +188,30 @@ def point_modal(omega, x0, n0, grid, L, N=None, deltan=0, c=None):
     x0 = util.asarray_1d(x0)
     x, y, z = util.as_xyz_components(grid)
 
+    if np.isscalar(N):
+        N = N * np.ones(3)
```

Comment:
`np.ones()` creates floating point values by default.
You should use `dtype=int` to get integers.

Then you can remove the `int()` from line 204.